frontend/components/molecules/Wysiwyg.js

```diff
@@ -0,0 +1,125 @@
+import autosize from 'autosize'
+import classNames from 'classnames'
+import { useTranslations } from 'next-intl'
+import { useEffect, useRef, useState } from 'react'
+import {
+  MdFormatBold,
+  MdFormatItalic,
+  MdStrikethroughS,
+  MdLink,
+  MdImage,
+  MdEdit,
+  MdPreview,
+} from 'react-icons/md'
+import bbcode from '../../libs/bbcode'
+import Button from '../atoms/Button'
+
+/**
+ *
+ * @param {HTMLTextAreaElement} input
+ * @param {string} type
+ */
+function insertAtCaret(input, type) {
+  const selected = input.value.slice(input.selectionStart, input.selectionEnd)
+
+  let rangeText = ''
+
+  switch (type) {
+    case 'b':
```

Comment:
Maybe extract it to a constant array and use `.contains()`

Reply:
Na there will be more complex tags in the near future, it is to change the text inserted not to check if it exists